include/ccl_defs.h

```diff
@@ -0,0 +1,21 @@
+/* @file
+ *
+ * Defines useful macros for compilation.
+ * */
+
+
+/* define marks for
+ * begin and end of declaration sections of header files for C++ compatibility */
+
+#ifdef __cplusplus
+
+#define CCL_BEGIN_DECLS extern "C" {
+#define CCL_END_DECLS }
```

Comment:
These macro names are confusing to me and generic. Maybe something that indicates they are for extern/C++ stuff?

Reply:
I learned of this trick / name from GTK:

https://code.woboq.org/gtk/include/errno.h.html#31

Reply:
cdefs.h defines similar macros but I *think* the file is not "portable".

Reply:
OK.